week-8/Homework/mandatory/3-dog-photo-gallery/script.js

```diff
@@ -0,0 +1,35 @@
+// Write a function that makes an API call to `https://dog.ceo/api/breeds/image/random`. It should trigger after clicking a button in your webpage. Every time the button is clicked it should append a new dog image to the DOM.
+
+// - Create an `index.html` file that will display your random image
+// - Add 2 `<button>` and 1 `<ul>` element, either in the HTML or through JavaScript
+// - When the button is clicked it should make an API call to `https://dog.ceo/api/breeds/image/random`
+// - After receiving the data, append to the `<ul>` a `<li>` that contains an `<img>` element with the dog image
+// - Incorporate error handling
+
+function getImageData() {
+  fetch("https://dog.ceo/api/breeds/image/random")
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      console.log(data)
+      newImage(data);
+    })
+    .catch((error)=>{
+      console.error('Something went wrong')
```

Comment:
This works really well for showing the developer the error information, but it doesn't really tell a normal user that anything has gone wrong. Can you think how to also let the user know what's happening / what to expect? Similarly for the humour question :)